File: src/lib/ua/ksn/geom/Point.cs
namespace ua.ksn.geom {

public struct Point {
   public int x,y;

   //public Point() { x=y=0; }
   public Point(int x, int y) { this.x = x; this.y = y; }
   public Point(Point p) { this.x = p.x; this.y = p.y; }

   public static bool operator !=(Point p1, Point p2) { return (p1.x != p2.x) || (p1.y != p2.y); }
   public static bool operator ==(Point p1, Point p2) { return (p1.x == p2.x) && (p1.y == p2.y); }

   public override bool Equals(object other) {
      if (!(other is Point))
         return false;
      return this == (Point)other;
   }

   public override int GetHashCode() {
      int sum = x + y;
      return sum * (sum + 1) / 2 + y;
   }

   public override string ToString() {
      return "[x=" + x + ", y=" + y + "]";
   }

#if WINDOWS_RT
   public static explicit operator Windows.Foundation.Point(Point self) { return new Windows.Foundation.Point() { X = self.x, Y = self.y }; }
#elif WINDOWS_FORMS
      ...
#endif
}
}

File: FastMines_core/src/fmg/common/geom/Point.java
package fmg.common.geom;

import java.io.Serializable;

public class Point implements Serializable {

    private static final long serialVersionUID = 1L;

    public int x;
    public int y;

    public Point() { x = y = 0; }
    public Point(int x, int y) { this.x = x; this.y = y; }
    public Point(Point p) { this.x = p.x; this.y = p.y; }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof Point))
            return false;
        Point p = (Point)other;
        return (x == p.x) && (y == p.y);
    }

    @Override
    public int hashCode() {
        int sum = x + y;
        return sum * (sum + 1) / 2 + y;
    }

    @Override
    public String toString() { return "{ x:" + x + ", y:" + y + " }"; }

}
